### DISCUSSION TITLE
cant compile delphi package for 11.3 the R part works ,D part fails

### DISCUSSION THREAD

**Discussion by MaheshNaikM** [graphics32/graphics32#266] [General] [answered]
dcc32 command line for "GR32_D.dpk"
  [dcc32 Fatal Error] GR32_Dsgn_Bitmap.pas(49): E2213 
  Bad packaged unit format: ..\..\GR32_R.dcp.GR32 - Expected version: 35.0, 
  Windows Unicode(x86) Found version: 34.0, Windows Unicode(x86)
Failed     
Please help... I dont know delphi..i want to take it to cbuilder...I was doing that with older 10.2 cbuilder and it works with rad studio 10.4.2

**Comment by andersmelander** [comment]
Did you read the help for this error message?
https://docwiki.embarcadero.com/RADStudio/Sydney/en/E2213_Bad_packaged_unit_format_%25s.%25s_(Delphi)

The error message even states what the problem is: The GR32_R package was compiled with a different version (version 34 = Delphi/CBuilder 10.4) than the one you are using now (version 35 = Delphi/CBuilder 11.0). So to the fix the problem you just need to rebuild the GR32_R package first.

**Reply by MaheshNaikM** [reply]
I have created a VM with only community edition Delphi 11.3 and trying to create the required packages and hpp files for c++(Not yet installed). I tried again with delphi and got the same  message...The hpp files are created in the R pass when i invoke the groupproj. The D package  gives the error...I only have the Delphi community edition installed on my Oracle VM.

**Reply by andersmelander** [reply]
I don't know how it's possible to get the wrong version when you only have one.

Regardless, the design-time package (GR32_D) depends on the run-time package (GR32_R), so you need to successfully build the run-time package first.

**Reply by MaheshNaikM** [reply]
I do not have any other programs/utilities on the VM. I tried again ...the required bad file is created in the bpl,dcp directories via tools option of delphi ....
If you advice i will create the files using a full trial version of rad studio 11  tomorrow..It is IST 10.48 p.m. now
the  .dcp is 1941 KB and .bpl is 1218 KB

**Reply by andersmelander** [reply]
I don't really have an explanation for the problem you're seeing or any suggestions for how to fix it. Sorry.

**Reply by MaheshNaikM** [reply]
it works with 10.4.2 nicely...Do i have to load a different graphics32 source for version 11?.I used the old version which i used for 10.4.2 community edition. i tried with full trial edition of studio v11 ...same error in d package. probably I donot understand how to compile the packages...I will read the delphi documentation...

**Reply by andersmelander** [reply, marked as answer]
> Do i have to load a different graphics32 source for version 11?

There are different package projects for Delphi 10.4 and Delphi 11 if that's what you mean.
The ones for Delphi 11.x are `Source\Packages\D110\GR32_D.dproj` and `...\GR32_R.dproj`.

The easiest way to load the packages is:

1. Load the group project `Graphics32_ packages.groupproj`.
2. Right-click on the **GR32_R** project and select **Build**. \
  ![billede](https://github.com/graphics32/graphics32/assets/18141921/474a62f7-969e-4907-98f5-047d806ea944)
3. Double click the **GR32_D** project to activate it.
3. Right-click on the **GR32_D** project and select **Install** (this will also do a build). \  
  ![billede](https://github.com/graphics32/graphics32/assets/18141921/30e09968-edda-4ab2-85f4-e8f5786f9df9)

**Comment by MaheshNaikM** [comment]
Thank you ..Thank you.. I downloaded a latest copy from github and did as you instructed. I got the  5 R package files and also the 5 D package files + the required .hpp files.  AS a curiosity i want to generate the .dcu filles ,I tried with the the option generate only .dcu files but i cant locate the files ?. ..I dont need them as i  do ,for this program, only vanilla .cpp using cbuilder.
